test/scala-mode-test.el

```diff
@@ -30,4 +30,41 @@ object Ensime {
     ('font-lock-keyword-face "K")
     ('font-lock-comment-face "O")
     ('font-lock-comment-delimiter-face "D")
+    ('font-lock-doc-face "U")
     (_ "?")))
+
+(ert-deftest smt:syntax-class-and-font-lock-test-1 ()
+  (smt:test
+   "val `tvw xyz/*` = `abc def/*` + 123 /* comment `abc` abc */ + 456"
+   "22203333333333301033333333333010222011022222220333330222011010222"
+   "KKK-VVVVVVVVVVV-K---------------CCC-DDDOOOOOOOOOOOOOOOOOOOO---CCC"))
+
+(ert-deftest smt:syntax-class-and-font-lock-test-2 ()
+  (smt:test
+   "val |+| = 123"
+   "2220333010222"
+   "KKK-VVV-K-CCC"))
+
+(ert-deftest smt:syntax-class-and-font-lock-test-3 ()
+  (smt:test
+   "val a_|+| = 123"
+   "222023333010222"
+   "KKK-VVVVV-K-CCC"))
+
+(ert-deftest smt:syntax-class-and-font-lock-test-4 ()
+  (smt:test
+   "val a = 123 /** hello */"
+   "222020102220111022222011"
+   "KKK-V-K-CCC-UUUUUUUUUUUU"))
+
+(ert-deftest smt:syntax-class-and-font-lock-test-5 ()
+  (smt:test
+   "val a = <td>hello</td>"
+   "2220201012212222211221"
+   "KKK-V-K---------------"))
+
+(ert-deftest smt:syntax-class-and-font-lock-test-6 ()
+  (smt:test
+   "// val |--| = 123"
+   "11022203333010222"
+   "DDDOOOOOOOOOOOOOO"))
```

Comment:
maybe add one that shows that the `<` in `:<:` is tagged as 3, but not in `Foo<T>`